apollo-federation/src/sources/connect/fetch_dependency_graph/mod.rs

```diff
@@ -757,6 +986,451 @@ mod tests {
+                                        ],
+                                        named_selections: vec![
+                                            // - Node's selection: a b { x z { three } } c: last
+                                            (name!("qax"), vec![Key::Field("qaax".to_string())]),
+                                            (
+                                                name!("baaz"),
+                                                vec![
+                                                    Key::Field("baz".to_string()),
+                                                    Key::Field("buzz".to_string()),
+                                                ],
+                                            ),
+                                        ],
+                                        tail_selection: Some((
+                                            name!("biz"),
+                                            connect::fetch_dependency_graph::PathTailSelection::CustomScalarPathSelection {
+                                                path_selection: connect::fetch_dependency_graph::PathSelection::Selection(connect::SubSelection {
+                                                    selections: vec![
+                                                        NamedSelection::Group(
+                                                            Alias { name: "blah".to_string() },
+                                                            connect::SubSelection {
+                                                                selections: vec![
+                                                                    NamedSelection::Field(None, "x".to_string(), None),
+                                                                    NamedSelection::Field(None, "y".to_string(), None)
+                                                                ],
+                                                                star: None
+                                                            })
+                                                    ],
+                                                    star: None
+                                                })
+                                            },
+                                        )),
+                                    },
+                            }),
+                        },
+                    ),
+                    &mut node,
+                )
+                .unwrap();
+
+            let source::fetch_dependency_graph::Node::Connect(result) = node else {
+                unreachable!()
+            };
+            assert_debug_snapshot!(result, @r###"
+            Node {
+                merge_at: [],
+                source_entering_edge: EdgeIndex(3),
+                field_response_name: "_merge_nested_path_test",
+                field_arguments: {},
+                selection: Some(
```

Comment:
how hard is it to make the snapshot use something like `to_string_pretty()` (which we'd need to write) so that these tests are possible to read?

Reply:
Done!